atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -51,7 +51,7 @@ public String getShortTableNameFromMappingTable(
             return unmappingCache.get(fullTableName, () -> {
                 SqlConnection conn = null;
                 try {
-                    conn = connectionSupplier.getNewUnsharedConnection();
+                    conn = connectionSupplier.get();
```

Comment:
If you close this connection after making the unmapper cache, the calling function cannot reuse the connection. This will make subsequent Db connections on this supplier fail. @clockfort for SA.